source/common/upstream/cluster_manager_impl.cc

```diff
@@ -416,7 +418,42 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
   // been setup for cross-thread updates to avoid needless updates during initialization. The order
   // of operations here is important. We start by initializing the thread aware load balancer if
   // needed. This must happen first so cluster updates are heard first by the load balancer.
-  auto cluster_data = active_clusters_.find(cluster.info()->name());
+  auto cluster_data = warming_clusters_.find(cluster.info()->name());
+  if (cluster_data != warming_clusters_.end()) {
+    if (cluster.info()->transportSocket().has_typed_config()) {
+      auto upstream_tls_context = MessageUtil::anyConvert<
+          envoy::extensions::transport_sockets::tls::v3::UpstreamTlsContext>(
+          cluster.info()->transportSocket().typed_config());
+      const auto& common_tls_context = upstream_tls_context.common_tls_context();
+
+      for (const auto& sds_secret_config :
+           common_tls_context.tls_certificate_sds_secret_configs()) {
+        auto& config_name = sds_secret_config.name();
+        auto& config_source = sds_secret_config.sds_config();
+        if (!secret_manager_.checkTlsCertificateEntityExists(config_source, config_name)) {
```

Comment:
Could you add the comment when `checkTlsCertificateEntityExists` could return false?

What is the recovery path to fix if it's temporary failure? An implicit retry? Redeliver the same SDS config or cluster config?

Reply:
I think that there is no secret entity should be treated as failed to initialize cluster, so that it should be warming forever. Do you have any idea about this to be more reliable?

Reply:
comment addition done.